### PR TITLE
Escape git fetch remote

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: services/migrations/gitea_uploader.go [hunk 1]
```diff
@@ -553,7 +553,7 @@ func (g *GiteaLocalUploader) updateGitForPullRequest(pr *base.PullRequest) (head
 			}
 
 			if ok {
-				_, _, err = git.NewCommand(g.ctx, "fetch", remote, pr.Head.Ref).RunStdString(&git.RunOpts{Dir: g.repo.RepoPath()})
+				_, _, err = git.NewCommand(g.ctx, "fetch", "--no-tags", "--", remote, pr.Head.Ref).RunStdString(&git.RunOpts{Dir: g.repo.RepoPath()})
```
Comment (zeripath) [inline services/migrations/gitea_uploader.go:556]: Are we sure we don't want tags here?

Comment (lafriks) [reply services/migrations/gitea_uploader.go:556]: I don't think pull request needs tags 🤔

Comment (zeripath) [reply services/migrations/gitea_uploader.go:556]: By default, any tag that points into the histories being fetched is
       also fetched; the effect is to fetch tags that point at branches that
       you are interested in. This default behavior can be changed by using
       the --tags or --no-tags options or by configuring remote.<name>.tagOpt.
       By using a refspec that fetches tags explicitly, you can fetch tags
       that do not point into branches you are interested in as well.

OKs.

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
-> #19490

**Comment by mweinelt** [comment]
Was a CVE requested for this issue? Because this looks exploitable.

**Comment by Li4n0** [comment]
> @mweinelt one was requested, pending assignment

It looks like there is an error in the [CVE](https://nvd.nist.gov/vuln/detail/CVE-2022-30781) description, 1.6.7 should be 1.16.7.

**Comment by 6543** [comment]
already reported

**Comment by mweinelt** [comment]
Will there be a patch for 1.15.11? And do you have an EOL policy somewhere?

**Comment by wxiaoguang** [comment]
https://github.com/go-gitea/gitea/blob/main/CONTRIBUTING.md#release-cycle

```
During a development cycle, we may also publish any necessary minor releases for the previous version. 
For example, if the latest, published release is v1.2, then minor changes for the previous release
 — e.g., v1.1.0 -> v1.1.1—are still possible.
```

Although it's not very obvious, it means that the latest stable releases would get updates implicitly. At the moment, the latest stable release is 1.16

**Comment by 6543** [comment]
feel free to cherry-pick the commit if you relay on 1.15 ... but you should really upgrade anyway

the build fails at https://github.com/go-gitea/gitea/pull/19728 ... so well it's EOL for sure!!!

**Pull request merged**